### PR TITLE
Add additional note about the Checkstyle memory issues

### PR DESCRIPTION
**Pull request by asodja**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -309,6 +309,9 @@ The following are the features that have been promoted in this Gradle release.
 
 Known issues are problems that were discovered post release that are directly related to changes made in this release.
 
+Projects with large source sets might experience out of memory errors due to changes to the [Checkstyle plugin](userguide/checkstyle_plugin.html):
```
Comment (ljacomet) [inline subprojects/docs/src/docs/release/notes.md:312]: This needs to go into the upgrade guide as a potential breaking change. There should be no content added to this section, it is a generated list of issues only.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -274,7 +274,7 @@ This flag may be useful in CI environments to record Gradle version information
 
 #### Checkstyle tasks use toolchains and execute in parallel by default
 
-The [Checkstyle plugin](userguide/checkstyle_plugin.html) now uses the Gradle worker API to run Checkstyle as an external worker process, so that multiple Checkstyle tasks may now run in parallel within a project. This can greatly increase overall build performance when several of these tasks exist within a single project. The memory used by the process is controlled via the `minHeapSize` and `maxHeapSize` properties.
+The [Checkstyle plugin](userguide/checkstyle_plugin.html) now uses the Gradle worker API to run Checkstyle as an external worker process, so that multiple Checkstyle tasks may now run in parallel within a project. This can greatly increase overall build performance when several of these tasks exist within a single project. In case of memory issues, you can adjust the memory used by the Checkstyle process via the `minHeapSize` and `maxHeapSize` properties on the Checkstyle task. By default the process will start with `maxHeapSize` of 512MB. We also recommend to use Checkstyle 9.3 or later.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/release/notes.md:277]: For info about fixing potential issues with this change, I would instead link to the upgrade guide.

**Review by ljacomet — CHANGES_REQUESTED** [review]
See below

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (asodja) [reply subprojects/docs/src/docs/release/notes.md]: Removed

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (asodja) [reply subprojects/docs/src/docs/release/notes.md]: Added link to upgrade guide.

**Review by ljacomet — APPROVED** [review]

**Pull request merged**